### PR TITLE
Fix PHP7.3 notices when non-existent coupon entered

### PR DESCRIPTION
**Pull request by scottcwilson**

### CODE REVIEW

File: includes/modules/order_total/ot_coupon.php [hunk 1]
```diff
@@ -170,17 +170,25 @@ function collect_posts() {
...
+    }
+
     if ((isset($_POST['dc_redeem_code']) && $_POST['dc_redeem_code'] != '') || (isset($discount_coupon->fields['coupon_code']) && $discount_coupon->fields['coupon_code'] != '')) {
       // set current Discount Coupon based on current or existing
-      if (isset($_POST['dc_redeem_code']) && $discount_coupon->fields['coupon_code'] == '') {
-        $dc_check = $_POST['dc_redeem_code'];
-      } else {
-        $dc_check = $discount_coupon->fields['coupon_code'];
-      }
+      if ($discount_coupon != null) {
```
Comment (drbyte) [inline includes/modules/order_total/ot_coupon.php]: Is the parent `if` needed anymore if we add these nested if/else stuff?

File: includes/modules/order_total/ot_coupon.php [hunk 2]
```diff
@@ -170,17 +170,25 @@ function collect_posts() {
...
+        if (isset($_POST['dc_redeem_code']) && $discount_coupon->fields['coupon_code'] == '') {
```
Comment (drbyte) [inline includes/modules/order_total/ot_coupon.php]: `$_POST['dc_redeem_code']` is always set, per line 160. Maybe this should be checking for `empty`?

Comment (drbyte) [reply includes/modules/order_total/ot_coupon.php]: Oh wait, maybe not if "removed".

File: includes/modules/order_total/ot_coupon.php (hunk 1 above)
Comment (scottcwilson) [reply includes/modules/order_total/ot_coupon.php]: I believe so.  This is pretty hairy logic that took a while to debug.

**Pull request merged**